lib/dice_page.dart

```diff
@@ -1,101 +1,128 @@
+        initialData: widget.bloc.defaultDice(),
+        builder: (
+          context,
+          snapshot,
+        ) {
+          return Container(
+            color: widget.bloc.dicesAreEqual ? Colors.green : Colors.red,
+            child: Center(
+              child: Column(
+                mainAxisAlignment: MainAxisAlignment.center,
+                children: <Widget>[
+                  Text(
+                    title,
+                    style: TextStyle(fontSize: Constants.titleFontSize, color: Colors.white,),
+                  ),
+                  Text(
+                    widget.bloc.dicesAreEqual ? "Dices Matched" : "Keep Trying",
                     style: TextStyle(
-                      color: Colors.black,
                       fontWeight: FontWeight.bold,
-                      fontSize: Constants.playButtonFontSize,
+                      color: Colors.black,
+                      fontSize: Constants.answerFontSize,
                     ),
                   ),
-                  style: ElevatedButton.styleFrom(
-                    shape: RoundedRectangleBorder(
-                      borderRadius: BorderRadius.circular(
-                        Constants.playButtonBorderRadius,
-                      ),
+                  Container(
+                    margin: EdgeInsets.all(
+                      Constants.dicesContainerMargin,
                     ),
-                    primary: Colors.blue,
-                    padding: EdgeInsets.fromLTRB(
-                      Constants.playButtonTopBottomPadding,
-                      Constants.playButtonLeftRightPadding,
-                      Constants.playButtonTopBottomPadding,
-                      Constants.playButtonLeftRightPadding,
+                    child: _buildImage(snapshot),
+                  ),
+                  Container(
+                    child: ElevatedButton(
+                      child: Text(
+                        'Good luck!',
+                        textAlign: TextAlign.center,
+                        style: TextStyle(
+                          color: Colors.black,
+                          fontWeight: FontWeight.bold,
+                          fontSize: Constants.playButtonFontSize,
+                        ),
+                      ),
+                      style: ElevatedButton.styleFrom(
+                        shape: RoundedRectangleBorder(
+                          borderRadius: BorderRadius.circular(
+                            Constants.playButtonBorderRadius,
+                          ),
+                        ),
+                        primary: Colors.blue,
+                        padding: EdgeInsets.symmetric(
+                          horizontal: Constants.playButtonLeftRightPadding,
+                          vertical: Constants.playButtonTopBottomPadding,
+                        ),
+                      ),
+                      onPressed: () {
+                        widget.bloc.diceChanger();
```

Comment:
You can change this method name with a more descriptive like rollDice or something like that